interchain/tests/src/bin/setup_funds.rs

```diff
@@ -82,7 +82,7 @@ pub fn test_send_funds() -> AnyResult<()> {
     // )?;
 
     // Get the ibc client address
-    let remote_account = AbstractAccount::new(&abstr_stargaze, Some(remote_account_id.clone()));
+    let remote_account = AbstractAccount::new(&abstr_stargaze, remote_account_id.clone());
     let client = remote_account.proxy.config()?;
```

Comment:
For example I think this query will fail because the remote account doesn't have an address associated with it. It's purely constructed from its ID but not fetched from the version control. I think it makes sense to only allow `AbstractAccount` creation (outside this crate) from:

`AccountFactory::create_account()`
`VersionControl::get_account()`

Otherwise an account object can be created without it having valid state.

Reply:
> For example I think this query will fail because the remote account doesn't have an address associated with it. It's purely constructed from its ID but not fetched from the version control.

`AbstractAccount::new` currently fetches account addresses from the version control

> I think it makes sense to only allow `AbstractAccount` creation (outside this crate) from:
> 
> `AccountFactory::create_account()` `VersionControl::get_account()`
> 
> Otherwise an account object can be created without it having valid state.

Proxy/Manager - yes, account - not really